### PR TITLE
[Tests-only] adjusted combined given-when steps inside Provisioning

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
Combined `given-when` steps splitted into seperate functions and resonable `check-success` assertion added on **Given** steps.

## Related Issue
- Part of https://github.com/owncloud/QA/issues/635

## Motivation and Context
- see https://github.com/owncloud/QA/issues/635

## How Has This Been Tested?

- :robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/Provisioning.php [hunk 1]
```diff
@@ -1213,6 +1350,23 @@ public function adminChangesTheDisplayOfUserUsingTheProvisioningApi(
...
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function adminHasChangedTheDisplayOfUserUsingTheProvisioningApi(
+		$user, $displayname
+	) {
+		$this->adminChangesTheDisplayNameOfUserUsingKey(
+			$user, 'display', $displayname
+		);
```
Comment (HariBhandari07) [inline tests/acceptance/features/bootstrap/Provisioning.php]: I think some kind of check is required here

File: tests/acceptance/features/bootstrap/Provisioning.php [hunk 2]
```diff
@@ -1291,6 +1461,24 @@ public function userChangesTheDisplayOfUserUsingTheProvisioningApi(
...
+	 */
+	public function userHasChangedTheDisplayOfUserUsingTheProvisioningApi(
+		$requestingUser, $targetUser, $displayName
+	) {
+		$this->userChangesTheDisplayNameOfUserUsingKey(
+			$requestingUser, $targetUser, 'display', $displayName
+		);
+		$this->theHTTPStatusCodeShouldBeSuccess();
+	}
+
```
Comment (HariBhandari07) [inline tests/acceptance/features/bootstrap/Provisioning.php]: This step seems a duplication of the above-given step you created. Also, check if the when steps were duplicated.

File: tests/acceptance/features/bootstrap/Provisioning.php (hunk 1 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/Provisioning.php]: check step is inside the `adminChangesTheDisplayNameOfUserUsingKey` function!!!

File: tests/acceptance/features/bootstrap/Provisioning.php (hunk 2 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/Provisioning.php]: Actually they are different. There are two cases:
- [x] admin changes display name of user
- [x] user changes display name of user

Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/Provisioning.php]: Also, display name can be edited sending the key `display` or `displayname` to the API endpoint. First one is older and inconsistent and later one is newer and consistent key-name. There are steps for both!

**Comment by kiranparajuli589** [comment]
- We can use consistent key to edit user, so I've removed _Given_ steps to edit user display-name using  key `display`. 
- Also some _When_ steps to edit user display-name are modified

File: tests/acceptance/features/bootstrap/Provisioning.php [hunk 3]
```diff
@@ -1350,23 +1350,6 @@ public function adminChangesTheDisplayOfUserUsingTheProvisioningApi(
 		);
 	}
 
-	/**
-	 * @Given /^the administrator has changed the display of user "([^"]*)" to "([^"]*)"$/
```
Comment (HariBhandari07) [inline tests/acceptance/features/bootstrap/Provisioning.php]: @kiranparajuli589 are there any feature files steps that implement this function, just wondering?

**Comment by phil-davis** [comment]
Now I remember something about this display name stuff. I think that the API allows setting the display name with either parameter. So we do want to test that both of them work - because there might by clients that use one or other of these. We do not want one of the ways of using the API to accidentally stop working.

File: tests/acceptance/features/bootstrap/Provisioning.php (hunk 3 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/Provisioning.php]: NO :no_good_man:
Feature steps are also changed after this change!

File: tests/acceptance/features/apiProvisioning-v1/editUser.feature [hunk 4]
```diff
@@ -73,7 +73,7 @@ Feature: edit users
     And user "subadmin" has been made a subadmin of group "new-group"
     When user "subadmin" changes the quota of user "brand-new-user" to "12MB" using the provisioning API
     And user "subadmin" changes the email of user "brand-new-user" to "brand-new-user@example.com" using the provisioning API
-    And user "subadmin" changes the display of user "brand-new-user" to "Anne Brown" using the provisioning API
+    And user "subadmin" changes the display name of user "brand-new-user" to "Anne Brown" using the provisioning API
```
Comment (HariBhandari07) [inline tests/acceptance/features/apiProvisioning-v1/editUser.feature]: I think we need to revert it back to the `display of user` as we want to check for both `display` and `display name`

File: tests/acceptance/features/apiProvisioning-v2/editUser.feature [hunk 5]
```diff
@@ -73,7 +73,7 @@ Feature: edit users
     And user "subadmin" has been made a subadmin of group "new-group"
     When user "subadmin" changes the quota of user "brand-new-user" to "12MB" using the provisioning API
     And user "subadmin" changes the email of user "brand-new-user" to "brand-new-user@example.com" using the provisioning API
-    And user "subadmin" changes the display of user "brand-new-user" to "Anne Brown" using the provisioning API
+    And user "subadmin" changes the display name of user "brand-new-user" to "Anne Brown" using the provisioning API
```
Comment (HariBhandari07) [inline tests/acceptance/features/apiProvisioning-v2/editUser.feature]: same as above

File: tests/acceptance/features/bootstrap/Provisioning.php [hunk 6]
```diff
@@ -1353,23 +1353,6 @@ public function adminChangesTheDisplayOfUserUsingTheProvisioningApi(
...
-	 * @throws Exception
-	 */
-	public function adminHasChangedTheDisplayOfUserUsingTheProvisioningApi(
-		$user, $displayname
-	) {
-		$this->adminChangesTheDisplayNameOfUserUsingKey(
-			$user, 'display', $displayname
-		);
-	}
-
```
Comment (HariBhandari07) [inline tests/acceptance/features/bootstrap/Provisioning.php]: same as above

File: tests/acceptance/features/bootstrap/Provisioning.php [hunk 7]
```diff
@@ -1440,48 +1423,6 @@ public function userHasChangedTheDisplayNameOfUserUsingTheProvisioningApi(
...
-		$this->userChangesTheDisplayNameOfUserUsingKey(
-			$requestingUser, $targetUser, 'display', $displayName
-		);
-	}
-
-	/**
-	 * @Given /^user "([^"]*)" has changed the display of user "([^"]*)" to "([^"]*)"$/
-	 *
-	 * @param string $requestingUser
-	 * @param string $targetUser
```
Comment (HariBhandari07) [inline tests/acceptance/features/bootstrap/Provisioning.php]: same as above

Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/Provisioning.php]: done!

File: tests/acceptance/features/bootstrap/Provisioning.php (hunk 6 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/Provisioning.php]: unneeded, not used anywhere!

File: tests/acceptance/features/apiProvisioning-v2/editUser.feature (hunk 5 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/apiProvisioning-v2/editUser.feature]: done!

File: tests/acceptance/features/apiProvisioning-v1/editUser.feature (hunk 4 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/apiProvisioning-v1/editUser.feature]: done!

**Review by haribhandari07 — COMMENTED** [review]
LGTM

File: .jshintrc [hunk 8]
```diff
@@ -15,6 +15,9 @@
 	"indent": 4,
 	"browser": true,
 	"laxbreak": true,
+	"esversion" : 8,
+	"node": true,
+	"asi": true,
```
Comment (phil-davis) [inline .jshintrc]: why is this added?

**Review by phil-davis — COMMENTED** [review]
Please remove the "bonus" `.jshintrc` lines and squash  the commits

**Review by phil-davis — APPROVED** [review]
